src/targets/gpu/device/softmax.cpp

```diff
@@ -20,34 +20,58 @@ void softmax(hipStream_t stream, const argument& result, const argument& arg, in
     migraphx::shape batch_shape{result.get_shape().type(), batch_lens};
 
     hip_visit_all(result, arg, batch_shape)([&](auto output, auto input, auto batch) {
-        const index_int max_block_size = 256;
+        const index_int max_block_size = 128;
```

Comment:
This may not be a good choice when the reduction axis is very large. If this works better in other models that use softmax(such as resnet50 and inception) then we can keep it for now and then improve these choices when we convert this to the jit model.